experimental/devices/pca9685/pins.go

```diff
@@ -0,0 +1,115 @@
+package pca9685
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/physic"
+	gpiopin "periph.io/x/periph/conn/pin"
+)
+
+const (
+	dutyMax gpio.Duty = math.MaxUint16
+)
+
+type pin struct {
+	dev     *Dev
+	channel int
+}
+
+// CreatePin creates a gpio handle for the given channel.
+func (d *Dev) CreatePin(channel int) gpio.PinIO {
```

Comment:
Check if the channel is reasonable?

Optional: If you always know there's 16, you could pre-allocate them inside Dev.

Reply:
I didn't want to eagerly create and register pins as I did for mcp23 because there are surely use-cases without them. Checking for the channel makes sense though.

Reply:
added a check here: https://github.com/google/periph/pull/437/commits/a682bff12501065e21d972c8a9dc029f46777a36#diff-261ed4cedecd0dbbee9594b52f8274b7R29